core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -70,39 +81,23 @@ protected TableScan newRefinedScan(
   @Override
   public CloseableIterable<FileScanTask> planFiles(TableOperations ops, Snapshot snapshot,
                                                    Expression rowFilter, boolean caseSensitive, boolean colStats) {
-    LoadingCache<Integer, ManifestEvaluator> evalCache = Caffeine.newBuilder().build(specId -> {
-      PartitionSpec spec = ops.current().spec(specId);
-      return ManifestEvaluator.forRowFilter(rowFilter, spec, caseSensitive);
-    });
-
-    Iterable<ManifestFile> nonEmptyManifests = Iterables.filter(snapshot.manifests(),
-        manifest -> manifest.hasAddedFiles() || manifest.hasExistingFiles());
-    Iterable<ManifestFile> matchingManifests = Iterables.filter(nonEmptyManifests,
-        manifest -> evalCache.get(manifest.partitionSpecId()).eval(manifest));
-
-    Iterable<CloseableIterable<FileScanTask>> readers = Iterables.transform(
-        matchingManifests,
-        manifest -> {
-          ManifestReader reader = ManifestReader.read(
-              ops.io().newInputFile(manifest.path()),
-              ops.current().specsById());
-          PartitionSpec spec = ops.current().spec(manifest.partitionSpecId());
-          String schemaString = SchemaParser.toJson(spec.schema());
-          String specString = PartitionSpecParser.toJson(spec);
-          ResidualEvaluator residuals = ResidualEvaluator.of(spec, rowFilter, caseSensitive);
-          return CloseableIterable.transform(
-              reader.filterRows(rowFilter)
-                  .caseSensitive(caseSensitive)
-                  .select(colStats ? SCAN_WITH_STATS_COLUMNS : SCAN_COLUMNS),
-              file -> new BaseFileScanTask(file, schemaString, specString, residuals)
-          );
-        });
+    return planFiles(
+        ops, snapshot, rowFilter, caseSensitive, colStats, Predicates.alwaysTrue(), Predicates.alwaysTrue());
+  }
 
-    if (PLAN_SCANS_WITH_WORKER_POOL && snapshot.manifests().size() > 1) {
-      return new ParallelIterable<>(readers, ThreadPools.getWorkerPool());
-    } else {
-      return CloseableIterable.concat(readers);
-    }
+  protected CloseableIterable<FileScanTask> planFiles(TableOperations ops, Snapshot snapshot,
```

Comment:
I don't see much value in adding this method. I'd probably remove it and make separate calls to `ManifestGroup` from this class and from `IncrementalDataTableScan`.